### PR TITLE
aspcud: fix by updating the dependencies

### PR DESCRIPTION
**Pull request by Zimmi48**
###### Motivation for this change

This fixes issue #19950 by replacing the dependency of `aspcud` on standalone `gringo` by a dependency on `clingo` (which contains the `gringo` binary as part of a larger monolithic package).

Tagging @hakuch as the one who introduced `gringo`.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] OS X
   - [ ] Linux
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by sternenseemann** [comment]
`opam` works for me with this patch :+1:

File: pkgs/tools/misc/clingo/default.nix [hunk 1]
```diff
@@ -0,0 +1,37 @@
+{ stdenv, fetchFromGitHub,
+  bison, re2c, scons
+}:
+
+let
+  version = "5.1.0";
```
Comment (sternenseemann) [inline pkgs/tools/misc/clingo/default.nix]: Wouldn't it be better if this was inside of `mkDerivation`?

File: pkgs/tools/misc/clingo/default.nix [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+  installPhase = ''
+    mkdir -p $out/bin
+    cp build/release/{gringo,clingo,reify,lpconvert} $out/bin/
+  '';
+
+  meta = with stdenv.lib; {
+    description = "A grounder and solver for logic programs.";
+    homepage = http://potassco.org;
+    platforms = platforms.linux;
+    maintainers = [ maintainers.hakuch ];
```
Comment (sternenseemann) [inline pkgs/tools/misc/clingo/default.nix]: Should be `stdenv.lib.maintainers.hakuch`.

**Comment by Profpatsch** [comment]
Looks good, thanks!

**Comment by hakuch** [comment]
I was on vacation, so I missed most of this. Thanks to all of you for clearly documenting and taking care of all this so that we can all continue to use `opam`!

**Comment by sternenseemann** [comment]
Has anybody tested running `opam` after the update of `aspcud`?

`aspcud` doesn't seem to work for me:

```
[WARNING] External solver failed with inconsistent return value. Request saved to
          "/home/lukas/.opam/log/solver-error-24753-1.cudf"
[ERROR] External solver failure, please fix your installation and check /home/lukas/.opam/config
        and variable $OPAMEXTERNALSOLVER.
        You may also retry with option --use-internal-solver
```

Has anybody got a working `opam`?

**Comment by hakuch** [comment]
I was able to reproduce the problem seen by @sternenseemann. If we don't hear from @Zimmi48 in the next couple of days than I'll prepare a PR that reverts this change.

**Comment by sternenseemann** [comment]
@hakuch Reverting will be no good, the build was broken back then…

**Comment by Zimmi48** [comment]
@hakuch If you revert, make sure to fix at the same time the building problem that this PR was supposed to solve (#19950).
I don't have any knowledge about CUDF-solvers, thus I can't really investigate the problem that @sternenseemann is having.
My feeling is that while clingo is being maintained, aspcud isn't and it's a bit unfortunate to have to rely on an unmaintained program. The second alternative proposed by Opam is packup. We might have more luck with it but it does not look maintained either.
Maybe contacting the potassco project (developers of gringo/clingo/aspcud) or the opam team could be useful to get a better understanding of what's going on.
Then when we look at [this page](http://opam.ocaml.org/doc/Specifying_Solver_Preferences.html#Yestherearedifferentversionsoftheuserpreferencelanguage), we actually understand that all these solvers were intensively developed between 2010 and 2012, less so since then.

**Comment by hakuch** [comment]
@Zimmi48 I'm in the process of reverting and fixing the compilation error of gringo. I'll have a PR up shortly.

The issue of opam's dependency on aspcud is a bit of a pain point in the OCaml community. As a user (and not a contributor), I'm eager for the situation to be better there as well.

**Comment by hakuch** [comment]
I have a PR for this at https://github.com/NixOS/nixpkgs/pull/20365.

**Pull request merged**